packages/InteractiveProfilingTool-Core.package/IPTMessageTallyWrapper.class/instance/getIncrementalGarbageCollectionStats.st

```diff
@@ -1,21 +1,8 @@
-information extraction
+gc stats access
 getIncrementalGarbageCollectionStats
 
-	| gcStats incrementalAverageTime incrementalCount incrementalRelativeTime incrementalTime uptime |
-	
-	gcStats := self messageTally gcStats.
-	uptime := self messageTally time.
-					
-	incrementalCount := gcStats at: 9.
-	incrementalTime := gcStats at: 10.
-	incrementalRelativeTime := (incrementalTime / uptime * 100) roundTo: 1.
-		
-	incrementalAverageTime := (incrementalCount = 0)
-		ifTrue: [ nil ]
-		ifFalse: [ (incrementalTime / incrementalCount) asFloat roundTo: 1 ].
-
 	^ Dictionary newFrom: {
```

Comment:
@linusha @twollnik Erstmal danke, dass ihr euch Gedanken gemacht habt. 
Ich dachte wir haben uns damals geeinigt hier keine Auslagerungen vorzunehmen. Durch die Ausslagerung hat die API nun an Übersicht verloren, da diese mehr nur das macht was sie eigentlich soll. Zudem macht das Dictionary jetzt wenig sinn, da alles was dieses zurückgibt schon über die jeweiligen Methoden abgerufen werden kann. Also meiner Meinung nach sollten wir nur das Dictionary zurück geben und die Berechnungen nicht in öffentliche Methoden auslagern. Da die API einfach gehalten werden soll. Mit dem Auslagern der Berrechnungen in die Methoden verliert das System stark an Lesbarkeit und der Traid-Off ist zu groß.

Reply:
@Alexander-Dubrawski 
Wir haben deinen Kommentar gelesen und den Code noch einmal angeschaut. Insgesamt sind wir damit, wie es im Moment ist, zufrieden. Unsere Differenz hier läuft unserer Ansicht nach vor allem auf eine Stilfrage hinaus. Solltest Du mit dem aktuellen Ansatz sehr unzufrieden sein, steht es dir natürlich frei, den Code selbst anzupassen. 

Die Lesbarkeit des Systems leidet unserer Ansicht nach nicht sehr unter unseren Änderungen, da die neuen Methoden alle sehr übersichtlich sind und eindeutig benannt sind. Die vergrößerte Anzahl an Methoden wird unserer Ansicht nach durch die deutlich verbesserte Lesbarkeit der einzelnen Methoden gerechtfertigt.